src/repository_urls.rs

```diff
@@ -55,261 +48,150 @@ fn get_distro_name(sysinfo: &SystemInfo, distro: &str) -> Option<String> {
-                    get_distro_name(sysinfo, distro)?,
-                    Self::extract_snapshot_date(url)?,
-                )
-            }
-            Self::RV(url) => format!(
-                "{}/__linux__/{}/{}/src/contrib",
-                RV_BASE_URL,
-                get_distro_name(sysinfo, distro)?, //need to determine if RV will have same binary support/distro names
-                Self::extract_snapshot_date(url)?
-            ),
-            Self::Other(url) => {
-                //TODO: we cannot expect only snapshot date/latest pattern for other/RV/PRISM in the future
-                // but this unblocks some work right now
-                let snapshot_date = Self::extract_snapshot_date(url)?;
-                let trimmed_url = url.trim_end_matches(snapshot_date).trim_end_matches("/");
-                format!(
-                    "{}/__linux__/{}/{}/src/contrib",
-                    trimmed_url,
-                    get_distro_name(sysinfo, distro)?, //need to determine if RV will have same binary support/distro names
-                    snapshot_date
-                )
-            }
-        };
-
-        // binaries are only returned when query strings are set for the r version
-        let mut linux_url = Some(format!(
-            "{}/{}?r_version={}.{}",
-            dir_url, file_name, r_version[0], r_version[1]
+    // For x86_64, the path in which binaries are found switches after R/4.2
+    if r_version <= &[4, 2] {
+        return Some(format!(
+            "{}/bin/macosx/contrib/{}.{}/{}",
+            url, r_version[0], r_version[1], file_name
         ));
-
-        //arch is not necessarily required, but appended when present
-        if let Some(arch) = sysinfo.arch() {
-            linux_url = Some(format!("{}&arch={arch}", linux_url?));
-        };
-        linux_url
     }
 
-    fn extract_snapshot_date(url: &str) -> Option<&str> {
-        SNAPSHOT_RE
-            .captures(url)
-            .and_then(|c| c.get(0))
-            .map(|x| x.as_str())
+    Some(format!(
+        "{}/bin/macosx/big-sur-{arch}/contrib/{}.{}/{}",
+        url, r_version[0], r_version[1], file_name
+    ))
+}
+
+fn get_linux_url(
+    url: &str,
+    file_name: &str,
+    r_version: &[u32; 2],
+    sysinfo: &SystemInfo,
+    distro: &str,
+) -> Option<String> {
+    let mut parts = url.split('/').collect::<Vec<_>>();
+    // split on `/`` will split "https://..." as 3 parts. Want to ensure there is at least one more path element at end of url
+    if parts.len() < 4 {
+        return None;
     }
+    let edition = parts.pop()?;
+    let base_url = parts.join("/");
+
+    Some(format!(
+        "{}/__linux__/{}/{}/src/contrib/{}?r_version={}.{}{}",
+        base_url,
+        get_distro_name(sysinfo, distro)?,
+        edition,
+        file_name,
+        r_version[0],
+        r_version[1],
+        sysinfo.arch().map(|arch| format!("&arch={arch}")).unwrap_or_default()
+    ))
```

Comment:
same as above, it would be easier to read to have tmp variables above and put them directly in the format